src/Player.tsx

```diff
@@ -6,311 +6,277 @@ import { popNote } from './music-sheet';
-    function unpause() {
-        setPaused(false);
-        pausedRef.current = false;
-    }
+// https://tonejs.github.io/docs/14.7.77/type/Subdivision
+const notesDuration = [
+    "2n",
+    "4n",
+    "4n",
+    "4n",
+    "8n",
+    "8n",
+    "8n"
+];
 
-    function stop() {
-        setPaused(false);
-        pausedRef.current = false;
-        if (loop.current && loop.current.state === "started") {
-            loop.current.stop();
-            chordsLoop.current.stop();
-            Tone.getTransport().stop();
-            Tone.getTransport().position = 0;
-            Tone.getTransport().cancel();
-        }
-    }
 
 
-    function changeChordInstrument(instrument: string) {
-        pause();
-        currentChordInstrument.current = createInstrument(instrument);
-    }
-    function changeInstrument(instrument: string) {
-        pause();
-        currentInstrument.current = createInstrument(instrument);
-    }
-    function createInstrument(instrument: string) {
-        switch(instrument) {
-            case 'guitar-acoustic':
-                return create('guitar-acoustic');
-            case 'guitar-electric':
-                return create('guitar-electric');
-            case 'guitar-nylon':
-                return createGuitarNylon();
-            case 'piano':
-                return create('piano');
-            case 'harp':
-                return create('harp');
-            case 'violin':
-                return createViolin();
-            case 'bassoon':
-                return create('bassoon');
-            case 'saxophone':
-                return createSaxophone();
-            case 'xylophone':
-                return createXylophone();
-            case 'organ':
-                return create('organ');
-            case "Tone.Synth":
-                unpause();
-                return new Tone.PolySynth(Tone.Synth).toDestination();
-            case "Tone.AMSynth":
-                unpause();
-                return new Tone.PolySynth(Tone.AMSynth).toDestination();
-            case "Tone.FMSynth":
-                unpause();
-                return new Tone.PolySynth(Tone.FMSynth).toDestination();
-            case "Tone.MembraneSynth":
-                unpause();
-            return new Tone.MembraneSynth().toDestination();
-            case "Tone.MetalSynth":
-                unpause();
-                return new Tone.MetalSynth().toDestination();
-            case "Tone.MonoSynth":
-                unpause();
-                return new Tone.MonoSynth().toDestination();
-            case "Tone.PluckSynth":
-                unpause();
-                return new Tone.PluckSynth().toDestination();
-            case "Tone.PolySynth":
-                unpause();
-                return new Tone.PolySynth().toDestination();
-        }
-        
-    }
+function Player() {
+    const notes = useStore(musicSheet, (state) => state.notes.first(10))
+    const notesRef = useRef(notes);
+    notesRef.current = notes;
```

Comment:
Using a reference on the note list is needed here because it's used in callback that must not be regenerated when the list changes.